### PR TITLE
Add Grid Schema and TnT Records

### PR DESCRIPTION
**Pull request by dnewh**
This adds functionality to the Asset Track client to create and manage Track and Trace Records and seed Schemas and Pike Agents. It also updates the styling and copy in the client to look more polished and reflect these changes.

### CODE REVIEW

File: track_and_trace/server/system/submit_utils.js [hunk 1]
```diff
@@ -49,48 +56,75 @@ const awaitServerPubkey = () => {
 }
 
 const encodeHeader = (signerPublicKey, batcherPublicKey, payload) => {
+  const GRID_NAMESPACE = hash(FAMILY_NAME,6) + '01'
```
Comment (dplumb94) [inline track_and_trace/server/system/submit_utils.js]: This would be the schema namespace, since the Grid namespace is just `621dee`

File: track_and_trace/asset_client/src/services/payloads.js [hunk 2]
```diff
@@ -14,141 +14,140 @@
...
-        [actionKey]: action.proto.create(action.xform(actionData))
-      }).finish()
-    case 'Pike':
-      payload = PikePayload.encode({
-        action: PikePayload.Action[action.enum],
-        timestamp: Math.floor(Date.now() / 1000),
-        [actionKey]: action.proto.create(action.xform(actionData))
-      }).finish()
-  }
+// 'use strict'
```
Comment (dplumb94) [inline track_and_trace/asset_client/src/services/payloads.js]: It looks like this commit is just commenting out code

File: track_and_trace/asset_client/src/utils/addressing.js [hunk 3]
```diff
@@ -68,8 +68,13 @@ module.exports = {
     makePropertyPageAddresses(recordId, propertyNames, currentPage) {
         return propertyNames.map((name) => TNT_NAMESPACE + TNT_PROPERTY_SUBSPACE + hash(recordId, 36) + hash(name, 22) + encodePage(currentPage))
     },
+<<<<<<< HEAD
```
Comment (dplumb94) [inline track_and_trace/asset_client/src/utils/addressing.js]: merge conflict committed

File: track_and_trace/asset_client/src/services/auth.js [hunk 4]
```diff
@@ -102,7 +102,8 @@ const AuthService = {
     },
 
     getUserData: () => new Promise((resolve, reject) => {
-        let userStr = _localStoreGet(STORE_USER)
+        let userStr = localStorage.getItem(`${AuthService.namespace}/${STORE_USER}`)
+        debugger;
```
Comment (dplumb94) [inline track_and_trace/asset_client/src/services/auth.js]: remove debugger

File: track_and_trace/asset_client/src/services/records.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+'use strict'
```
Comment (dplumb94) [inline track_and_trace/asset_client/src/services/records.js]: This commit does not do what the description says

File: track_and_trace/asset_client/src/services/auth.js [hunk 6]
```diff
@@ -231,17 +232,16 @@ const AuthService = {
 
     /**
      * Creates a user then submits a transaction to the blockchain
-     * 
+     *
      * The function is a (Signer) => Promise, where the promise is resolved when the transaction completes
      */
     createUser: (user, submitTransactionFn) => {
-        let userCreate = pluck(user, 'username', 'password', 'email', 'name')
+        let userCreate = pluck(user, 'password', 'email')
```
Comment (dplumb94) [inline track_and_trace/asset_client/src/services/auth.js]: is the database also updated to reflect this?

File: track_and_trace/asset_client/src/views/add_asset_form.js [hunk 7]
```diff
@@ -134,81 +107,42 @@ const AddAssetForm = {
...
+    stringValue: state.serialNumber
+  },
+  {
+    name: 'type',
+    stringValue: state.type,
+    dataType: PropertyDefinition.DataType.STRING
+  },
+  {
+    name: 'weight',
+    numberValue: (state.weight ? parsing.toInt(state.weight) : 0),
```
Comment (dplumb94) [inline track_and_trace/asset_client/src/views/add_asset_form.js]: It should not be possible to submit the transaction if the weight or location is not set. It is possible for an asset to actually validly have a weight of 0 or a location of 0,0, so setting it to zero is not correct.

File: track_and_trace/docker-compose.yaml [hunk 8]
```diff
@@ -86,29 +86,6 @@ services:
       - 'no_proxy=rest-api,server,eth0,validator,${no_proxy}'
     entrypoint: node index.js
 
-  ledger-sync:
```
Comment (dplumb94) [inline track_and_trace/docker-compose.yaml]: The code should be removed too

File: track_and_trace/asset_client/utils/submit_utils.js [hunk 9]
```diff
@@ -0,0 +1,342 @@
+/**
```
Comment (dplumb94) [inline track_and_trace/asset_client/utils/submit_utils.js]: This looks like a lot of duplicate code? Why doesn't it use the existing transaction submission module?

Comment (dnewh) [reply track_and_trace/asset_client/utils/submit_utils.js]: the existing one utilizes mithril requests which aren't available when this script is run.

File: track_and_trace/asset_client/src/services/auth.js (hunk 6 above)
Comment (dnewh) [reply track_and_trace/asset_client/src/services/auth.js]: Yes, in a previous PR.

File: track_and_trace/asset_client/src/views/add_asset_form.js (hunk 7 above)
Comment (dnewh) [reply track_and_trace/asset_client/src/views/add_asset_form.js]: updated

File: track_and_trace/asset_client/src/utils/addressing.js [hunk 10]
```diff
@@ -32,15 +50,27 @@ module.exports = {
...
     makeOrganizationAddress(organizationId) {
         return PIKE_NAMESPACE + PIKE_ORG_SUBSPACE + hash(organizationId, 62)
     },
+    makeSchemaAddress(schemaName) {
+        return GRID_NAMESPACE + GRID_SCHEMA_SUBSPACE + hash(schemaName, 62)
+    },
+    makeRecordAddress(recordId) {
+        return TNT_NAMESPACE + TNT_RECORD_SUBSPACE + hash(recordId, 62)
+    },
+    makePropertyAddresses(recordId, propertyNames) {
```
[the remark this answers is not in the document]
Comment (dnewh) [reply track_and_trace/asset_client/src/utils/addressing.js]: updated

File: track_and_trace/asset_client/src/utils/addressing.js [hunk 11]
```diff
@@ -51,10 +81,17 @@ module.exports = {
...
+        namespace: GRID_NAMESPACE
+    },
     sabreFamily: {
         name: SABRE_FAMILY_NAME,
         version: SABRE_FAMILY_VERSION,
     },
+    propertyAddressPrefix: TNT_NAMESPACE + TNT_PROPERTY_SUBSPACE,
     agentAddressPrefix: PIKE_NAMESPACE + PIKE_AGENT_SUBSPACE,
     organizationAddressPrefix: PIKE_NAMESPACE + PIKE_ORG_SUBSPACE,
+    schemaAddressPrefix: GRID_NAMESPACE + GRID_SCHEMA_SUBSPACE,
```
[the remark this answers is not in the document]
Comment (dnewh) [reply track_and_trace/asset_client/src/utils/addressing.js]: updated

File: track_and_trace/protos/track_and_trace_state.proto [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+syntax = "proto3";
+
+import "schema_state.proto";
+
+message Property {
```
[the remark this answers is not in the document]
Comment (dnewh) [reply track_and_trace/protos/track_and_trace_state.proto]: updated

File: track_and_trace/server/scripts/seed_core_types.js [hunk 13]
```diff
@@ -34,7 +34,22 @@ const types = require(`./${DATA}`)
...
+          properties: [...type.properties]
+        })
+      })
+    })
+
+    const txns = typePayloads.map(payload => createTxn(payload))
+    return submitTxns(txns, signer, signerPublicKey)
+  })
   .catch(err => {
+    console.log("error in protos compile")
```
[the remark this answers is not in the document]
Comment (dnewh) [reply track_and_trace/server/scripts/seed_core_types.js]: updated

File: track_and_trace/asset_client/src/utils/addressing.js [hunk 14]
```diff
@@ -1,15 +1,23 @@
...
+const TNT_PROPERTY_SUBSPACE = 'ea'
+const TNT_FAMILY_VERSION = '1.0'
 
 const PIKE_FAMILY_NAME = 'pike'
 const PIKE_FAMILY_VERSION = '0.1'
 const PIKE_NAMESPACE = 'cad11d'
 const PIKE_AGENT_SUBSPACE = '00'
 const PIKE_ORG_SUBSPACE = '01'
 
+const GRID_FAMILY_NAME = 'grid_schema'
```
Comment (dplumb94) [inline track_and_trace/asset_client/src/utils/addressing.js]: This is the Grid Schema family name

File: track_and_trace/asset_client/src/utils/addressing.js [hunk 15]
```diff
@@ -1,15 +1,23 @@
...
+const GRID_FAMILY_VERSION = '1.0'
```
Comment (dplumb94) [inline track_and_trace/asset_client/src/utils/addressing.js]: This is the Grid Schema family version

File: track_and_trace/asset_client/src/utils/addressing.js (hunk 14 above)
Comment (dnewh) [reply track_and_trace/asset_client/src/utils/addressing.js]: updated

File: track_and_trace/asset_client/src/utils/addressing.js (hunk 15 above)
Comment (dnewh) [reply track_and_trace/asset_client/src/utils/addressing.js]: updated

File: track_and_trace/asset_client/utils/submit_utils.js (hunk 9 above)
Comment (dplumb94) [reply track_and_trace/asset_client/utils/submit_utils.js]: Ok. Lets put this in the scripts directory to make it clear that it is only for the scripts.

File: track_and_trace/asset_client/scripts/seed_core_types.js [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+        return createOrganization(org, signer)
+      })
+      .then((signer) => {
+        return updateAgent(org, signer, ['admin', 'can_create_schema', 'can_update_schema'])
+      })
+      .then((signer) => {
+        return createSchemas(org, signer)
+      })
+      .catch((err) => {
+        console.log(err)
```
Comment (dplumb94) [inline track_and_trace/asset_client/scripts/seed_core_types.js]: should probably be `console.error`

File: track_and_trace/server/scripts/seed_core_types.js [hunk 17]
```diff
@@ -34,7 +34,22 @@ const types = require(`./${DATA}`)
 protos.compile()
   .then(awaitServerPubkey)
   .then(batcherPublicKey => getTxnCreator(null, batcherPublicKey))
+  .then(({createTxn, signer, signerPublicKey}) => {
```
Comment (dplumb94) [inline track_and_trace/server/scripts/seed_core_types.js]: Where is this script used? We shouldn't be doing this stuff from the server anymore

File: track_and_trace/asset_client/utils/submit_utils.js [hunk 18]
```diff
@@ -0,0 +1,338 @@
...
+
+  return submitTxns(txns, signer)
+}
+
+const createSchemaTransaction = (schema, signer) => {
+  if (!signer) {
+    console.error('a signer must be provided')
+  }
+
+  let schemaCreate= protos.SchemaCreateAction.create({
```
Comment (dplumb94) [inline track_and_trace/asset_client/utils/submit_utils.js]: Looks like some strange formatting. I suggest running eslint on this file

File: track_and_trace/asset_client/src/services/records.js (hunk 5 above)
Comment (dnewh) [reply track_and_trace/asset_client/src/services/records.js]: updated

File: track_and_trace/docker-compose.yaml (hunk 8 above)
Comment (dnewh) [reply track_and_trace/docker-compose.yaml]: updated

File: track_and_trace/server/scripts/seed_core_types.js (hunk 17 above)
Comment (dnewh) [reply track_and_trace/server/scripts/seed_core_types.js]: updated

File: track_and_trace/server/system/submit_utils.js (hunk 1 above)
Comment (dnewh) [reply track_and_trace/server/system/submit_utils.js]: updated

File: track_and_trace/asset_client/utils/submit_utils.js (hunk 18 above)
Comment (dnewh) [reply track_and_trace/asset_client/utils/submit_utils.js]: updated

File: track_and_trace/server/scripts/seed_core_types.js (hunk 17 above)
Comment (dnewh) [reply track_and_trace/server/scripts/seed_core_types.js]: updated

File: track_and_trace/asset_client/scripts/seed_core_types.js (hunk 16 above)
Comment (dnewh) [reply track_and_trace/asset_client/scripts/seed_core_types.js]: updated

File: track_and_trace/asset_client/src/views/add_asset_form.js [hunk 19]
```diff
@@ -18,7 +18,7 @@
 const m = require('mithril')
 
 const api = require('../services/api')
-const payloads = require('../services/payloads')
+// const payloads = require('../services/payloads')
```
Comment (agunde406) [inline track_and_trace/asset_client/src/views/add_asset_form.js]: Remove commented out code

File: track_and_trace/asset_client/src/views/asset_detail.js [hunk 20]
```diff
@@ -62,39 +62,39 @@ const TransferDropdown = {
...
-                    _submitProposal(record, ROLE_TO_ENUM[role], agent.key)
-                      .then(onsuccess)
-                  }
-                }
-              }, m('span.text-truncate',
-                   truncate(agent.name, { length: 32 }),
-                   (proposal ? ' \u2718' : '')))
-            ]
-          })))
+      // m('.dropdown',
```
Comment (agunde406) [inline track_and_trace/asset_client/src/views/asset_detail.js]: Remove commented out code

File: track_and_trace/ledger_sync/subscriber/protos.js [hunk 21]
```diff
@@ -60,39 +60,24 @@ const compile = () => {
       'LatLong',
       'PropertyValue'
     ]),
-    loadProtos('track_and_trace_agent.proto', [
```
[the remark this answers is not in the document]
Comment (dnewh) [reply track_and_trace/ledger_sync/subscriber/protos.js]: fixed

File: track_and_trace/asset_client/src/views/add_asset_form.js (hunk 19 above)
Comment (dnewh) [reply track_and_trace/asset_client/src/views/add_asset_form.js]: fixed

File: track_and_trace/asset_client/src/views/asset_detail.js (hunk 20 above)
Comment (dnewh) [reply track_and_trace/asset_client/src/views/asset_detail.js]: fixed

File: track_and_trace/server/system/submit_utils.js [hunk 22]
```diff
@@ -19,23 +19,30 @@
...
+  BatchList
 } = require('sawtooth-sdk/protobuf')
 const protos = require('../blockchain/protos')
 
 const FAMILY_NAME = 'grid_track_and_trace'
 const FAMILY_VERSION = '1.0'
-const NAMESPACE = 'a43b46'
 
-const SERVER = process.env.SERVER || 'http://localhost:3000'
+const SERVER = 'http://tnt-asset-client:8021'
```
Comment (peterschwarz) [inline track_and_trace/server/system/submit_utils.js]: This shouldn't be locked to this URL, as it is very specific to the docker compose file.  It will break if used in any other context.

File: track_and_trace/asset_client/scripts/utils/submit_utils.js [hunk 23]
```diff
@@ -0,0 +1,338 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ----------------------------------------------------------------------------
+ */
+'use strict'
+
+const _ = require('lodash')
+const sjcl = require('sjcl')
+const request = require('request-promise-native')
```
Comment (peterschwarz) [inline track_and_trace/asset_client/scripts/utils/submit_utils.js]: Why the use of this, versus using the existing mithril request function?

File: track_and_trace/asset_client/src/services/parsing.js [hunk 24]
```diff
@@ -18,7 +18,8 @@
 
 const _ = require('lodash')
 const moment = require('moment')
-const { FLOAT_PRECISION } = require('./payloads')
+
+const FLOAT_PRECISION = 1000000
```
Comment (peterschwarz) [inline track_and_trace/asset_client/src/services/parsing.js]: There are a lot of deletions in this commit, but I'm unsure based on the commit message as to why

File: track_and_trace/asset_client/scripts/utils/submit_utils.js (hunk 23 above)
Comment (dnewh) [reply track_and_trace/asset_client/scripts/utils/submit_utils.js]: At this point the script is running before the window object is loaded which is required to use the mithril requests.

Comment (peterschwarz) [reply track_and_trace/asset_client/scripts/utils/submit_utils.js]: Ah, that makes sense

File: track_and_trace/server/system/submit_utils.js (hunk 22 above)
Comment (dnewh) [reply track_and_trace/server/system/submit_utils.js]: Updated

File: track_and_trace/asset_client/src/services/parsing.js (hunk 24 above)
Comment (dnewh) [reply track_and_trace/asset_client/src/services/parsing.js]: updated commit so changes updating the deleted code are part of this commit now.

File: track_and_trace/asset_client/utils/submit_utils.js (hunk 9 above)
Comment (dnewh) [reply track_and_trace/asset_client/utils/submit_utils.js]: updated

File: track_and_trace/asset_client/src/views/list_assets.js [hunk 25]
```diff
@@ -33,14 +33,17 @@ const AssetList = {
...
 
     const refresh = () => {
-      api.get('records?recordType=asset').then((records) => {
-        vnode.state.records = records
+      m.request({
+        method: 'GET',
+        url: '/grid/record'
+      })
+      .then((result) => {
+        vnode.state.records = result
```
Comment (eloaverona) [inline track_and_trace/asset_client/src/views/list_assets.js]: You should probably filter the result here so that it only shows records that have schema === 'asset'. Otherwise, when we have "fish" records it will also show on this list.

Comment (dnewh) [reply track_and_trace/asset_client/src/views/list_assets.js]: Good call. Updated.

File: track_and_trace/asset_client/src/views/dashboard.js [hunk 26]
```diff
@@ -27,39 +27,31 @@ const Dashboard = {
...
+          'of the asset changes, these updates are submitted to the ',
+          'distributed ledger, providing an immutable and auditable ',
+          'history of the asset.'),
         m('p',
           'To use ',
           m('em', 'AssetTrack'),
-          ', create an account using the link in the navbar above. ',
-          'Once logged in, you will be able to add new assets to ',
-          'the blockchain and track them with data like temperature or ',
+          ', create an account  using the "Get Started" link in the navbar ',
```
Comment (dplumb94) [inline track_and_trace/asset_client/src/views/dashboard.js]: extra space

Comment (dnewh) [reply track_and_trace/asset_client/src/views/dashboard.js]: updated

File: track_and_trace/asset_client/src/views/agent_detail.js [hunk 27]
```diff
@@ -96,42 +96,6 @@ const editIcon = (obj, key) => {
   return forms.clickIcon('pencil', () => { obj[key] = !obj[key] })
 }
 
-// Edits a field in state
-const editField = (state, label, key) => {
```
Comment (agunde406) [inline track_and_trace/asset_client/src/views/agent_detail.js]: Can you add a comment about why this stuff was deleted to the commit. Does it belong in a different commit?

Comment (dnewh) [reply track_and_trace/asset_client/src/views/agent_detail.js]: Updated commit message
